boards/arm/colibri_imx7d_m4/board.h

```diff
@@ -0,0 +1,102 @@
+#define BOARD_WDOG_IRQ_NUM                    WDOG3_IRQn
+#define BOARD_WDOG_HANDLER                    WDOG3_Handler
+
+/* SEMA4 information for this board */
+#define BOARD_SEMA4_RDC_PDAP                  rdcPdapSemaphoreHs
+#define BOARD_SEMA4_CCM_CCGR                  ccmCcgrGateSemaHs
+#define BOARD_SEMA4_BASEADDR                  SEMA4
+#define BOARD_SEMA4_IRQ_NUM                   SEMA4_HS_M4_IRQn
+#define BOARD_SEMA4_HANDLER                   SEMA4_HS_M4_Handler
+
+/* GPT instance A information for this board */
+#define BOARD_GPTA_RDC_PDAP                   rdcPdapGpt3
+#define BOARD_GPTA_CCM_ROOT                   ccmRootGpt3
+#define BOARD_GPTA_CCM_CCGR                   ccmCcgrGateGpt3
+#define BOARD_GPTA_BASEADDR                   GPT3
+#define BOARD_GPTA_IRQ_NUM                    GPT3_IRQn
+#define BOARD_GPTA_HANDLER                    GPT3_Handler
+/* GPT instance B information for this board */
+#define BOARD_GPTB_RDC_PDAP                   rdcPdapGpt4
+#define BOARD_GPTB_CCM_ROOT                   ccmRootGpt4
+#define BOARD_GPTB_CCM_CCGR                   ccmCcgrGateGpt4
+#define BOARD_GPTB_BASEADDR                   GPT4
+#define BOARD_GPTB_IRQ_NUM                    GPT4_IRQn
+#define BOARD_GPTB_HANDLER                    GPT4_Handler
+
+/* GPIO information for this board */
+#define BOARD_GPIO_KEY_CCM_CCGR               ccmCcgrGateGpio2
+#define BOARD_GPIO_LED_RDC_PDAP               rdcPdapGpio1
+#define BOARD_GPIO_LED_CONFIG                 (&gpioLed)
+#define BOARD_GPIO_KEY_RDC_PDAP               rdcPdapGpio2
+#define BOARD_GPIO_KEY_CONFIG                 (&gpioKeyFunc1)
+#define BOARD_GPIO_KEY_IRQ_NUM                GPIO2_INT31_16_IRQn
+#define BOARD_GPIO_KEY_HANDLER                GPIO2_INT31_16_Handler
+
+/* Debug UART information for this board */
+#define BOARD_DEBUG_UART_RDC_PDAP             rdcPdapUart2
+#define BOARD_DEBUG_UART_CCM_ROOT             ccmRootUart2
+#define BOARD_DEBUG_UART_CCM_CCGR             ccmCcgrGateUart2
+#define BOARD_DEBUG_UART_BASEADDR             UART2
+#define BOARD_DEBUG_UART_IRQ_NUM              UART2_IRQn
+#define BOARD_DEBUG_UART_HANDLER              UART2_Handler
+
+/* MU information for this board*/
+#define BOARD_MU_HANDLER                      MU_M4_Handler
+#define BOARD_MU_IRQ_NUM                      MU_M4_IRQn
+#define BOARD_MU_BASE_ADDR                    MUB
+#define BOARD_MU_CCM_CCGR                     ccmCcgrGateMu
+#define BOARD_MU_RDC_PDAP                     rdcPdapMuB
+
+/* I2C information for this board */
+#define BOARD_I2C_RDC_PDAP                    rdcPdapI2c4
+#define BOARD_I2C_CCM_ROOT                    ccmRootI2c4
+#define BOARD_I2C_CCM_CCGR                    ccmCcgrGateI2c4
+#define BOARD_I2C_BASEADDR                    I2C4
+#define BOARD_I2C_IRQ_NUM                     I2C4_IRQn
+#define BOARD_I2C_HANDLER                     I2C4_Handler
+#define BOARD_I2C_FXAS21002_ADDR              (0x20)
+#define BOARD_I2C_FXOS8700_ADDR               (0x1E)
```

Comment:
What are all the macros in this file for? Most look like things we can get from dts